### ISSUE TITLE
individual name for places of a brand

### ISSUE THREAD

**Issue by tiptoptom** [streetcomplete/StreetComplete#5985]
**How to Reproduce**
Try to add a place in the places overlay of a brand with an individual name for example:
```
shop=supermarket
brand=EDEKA
brand:wikidata=Q701755
name=EDEKA Schreiber
```
At the moment you cannot set an individual name for a place of a brand in the places overlay.

---

Or try to add the brand tag in the places overlay to a place that already has an individual name:
```
shop=supermarket
name=EDEKA Schreiber
``` 
The name tag is currently overwritten and the following tagging is obtained:
```
shop=supermarket
brand=EDEKA
brand:wikidata=Q701755
name=EDEKA
```
**Expected Behavior**

1. For places of a brand there should still be field to enter an individual name or at least an option to open such a field. 
2. The name tag of a place shouldn't be overridden when a brand is set. It should also be displayed and editable.

**Versions affected**

59.2

**Issues that may have a certain relationship**
#4688
#5313
#5940

**Wiki**
https://wiki.openstreetmap.org/wiki/Key:brand#Examples

**Comment by westnordost** [comment]
If EDEKAs usually have each an own name, why is `name` specified at all for the EDEKA brand?

**Comment by westnordost** [comment]
I.e. the preset should rather add `brand=EDEKA` rather than `name=EDEKA` when `name` is always different.

**Comment by tiptoptom** [comment]
I don't know whether the name is always different in the case of Edeka or whether it is only occasionally like that. Nevertheless, there is also no option to enter or change the name.

**Comment by westnordost** [comment]
For SC it is not possible to "detect" that "EDEKA Schreiber" is actually **an** "EDEKA" and henceforth should not be overwritten by the preset.
So, the only solution is to always show and let the user edit the name field, which I would really like to avoid in the case of brands.

**Comment by tiptoptom** [comment]
Would it be a solution, if the name does not match NSI, not to overwrite it but to display it in a field. And if the name matches NSI, to hide the field and only make it visible via the three-dot menu.

**Comment by westnordost** [comment]
> Would it be a solution, if the name does not match NSI, not to overwrite it but to display it in a field.

That would be very weird. Imagine there was a completely different shop in there before, you select "EDEKA" and the old name is still in there!

Anyway, actually, what exactly were you doing, by the way? You added a new EDEKA?

**Comment by westnordost** [comment]
iD locks the `brand` field for EDEKA, but not the `name` field:

![imagen](https://github.com/user-attachments/assets/27518c45-0262-4a3f-bbec-7c8a43e320f2)

For ALDI, it looks different:

![imagen](https://github.com/user-attachments/assets/d10f98f3-fa0b-45d3-a94c-68f0408a7bb4)

**Comment by tiptoptom** [comment]
> > Would it be a solution, if the name does not match NSI, not to overwrite it but to display it in a field.
> 
> That would be very weird. Imagine there was a completely different shop in there before, you select "EDEKA" and the old name is still in the name field!

No, because in this case the “Is it a different place now?” window would open. If “Different place now” is then selected, the name must of course be overwritten. 

> Anyway, actually, what exactly were you doing, by the way? You added a new EDEKA?

I have selected the NSI preset for Edeka for an object with the following tags:
```
shop=supermarket
name=EDEKA Schreiber
```

**Comment by westnordost** [comment]
> I have selected the NSI preset for Edeka for an object with the following tags

But why? And why did you change it?

**Comment by tiptoptom** [comment]
> > I have selected the NSI preset for Edeka for an object with the following tags
> 
> But why? And why did you change it?

Because brand=EDEKA and brand:wikidata=Q701755 were missing.

**Comment by westnordost** [comment]
Anyway, the NSI presets for the screenshots above are

```json

"shop/supermarket/edeka-0a839e": {
  "name": "EDEKA",
  "locationSet": {"include": ["de"]},
  "icon": "maki-grocery",
  "geometry": ["point", "area"],
  "matchScore": 2,
  "imageURL": "https://graph.facebook.com/EDEKA/picture?type=large",
  "terms": ["e neukauf", "edeka", "edeka neukauf"],
  "fields": ["name", "brand", "{shop/supermarket}"],
  "tags": {"brand:wikidata": "Q701755", "shop": "supermarket"},
  "addTags": {
    "brand": "EDEKA",
    "brand:wikidata": "Q701755",
    "name": "EDEKA",
    "shop": "supermarket"
  }
},
"shop/supermarket/aldi-813759": {
  "name": "ALDI Nord (Deutschland)",
  "locationSet": {"include": ["de"], "exclude": ["aldi-sued.geojson"]},
  "icon": "maki-grocery",
  "geometry": ["point", "area"],
  "matchScore": 2,
  "imageURL": "https://commons.wikimedia.org/wiki/Special:FilePath/AldiNord-WorldwideLogo.svg",
  "terms": ["aldi nord"],
  "tags": {"brand:wikidata": "Q41171373", "shop": "supermarket"},
  "addTags": {
    "brand": "ALDI Nord",
    "brand:wikidata": "Q41171373",
    "name": "Aldi",
    "shop": "supermarket"
  }
},

```

In conclusion, I have no idea how iD is able to treat these two differently.

Maybe @kjonosm knows?

**Comment by westnordost** [comment]
> Because brand=EDEKA and brand:wikidata=Q701755 were missing.

But StreetComplete didn't display them as missing.

**Comment by Cj-Malone** [comment]
> In conclusion, I have no idea how iD is able to treat these two differently.

`preserveTags`

https://nsi.guide/index.html?t=brands&k=shop&v=supermarket&tt=edeka#edeka-0a839e

But `branch` is usually the real answer.

**Comment by tiptoptom** [comment]
> > Because brand=EDEKA and brand:wikidata=Q701755 were missing.
> 
> But StreetComplete didn't display them as missing.

Not directly, but you can see a difference.

![d97c63c7-6e09-4a9b-8ea4-68a352f29a56](https://github.com/user-attachments/assets/740eecc2-d15f-4251-9217-10dc3fd35ef7) 
![fde6c4be-461b-4305-b06b-318e7e1c78cc](https://github.com/user-attachments/assets/bd0b316f-bfa9-44df-b2cf-c023a71541e7)

**Comment by westnordost** [comment]
@Cj-Malone but where is `preserveTags`? I don't see it in the [presets.json](https://raw.githubusercontent.com/osmlab/name-suggestion-index/refs/heads/main/dist/presets/nsi-id-presets.json)

**Comment by Cj-Malone** [comment]
It's in [nsi.json](https://raw.githubusercontent.com/osmlab/name-suggestion-index/refs/heads/main/dist/nsi.json)

**Comment by westnordost** [comment]
Hmm, StreetComplete and osmfeatures doesn't use that file at all. It seems to be somewhat of the [`nsi-id-presets.json`](https://raw.githubusercontent.com/osmlab/name-suggestion-index/refs/heads/main/dist/presets/nsi-id-presets.json) but in a different format. The entry for EDEKA in nsi.json is:

```json
{
  "displayName": "EDEKA",
  "id": "edeka-0a839e",
  "locationSet": {"include": ["de"]},
  "matchNames": ["e neukauf", "edeka neukauf"],
  "preserveTags": ["^name"],
  "tags": {"brand": "EDEKA", "brand:wikidata": "Q701755", "name": "EDEKA", "shop": "supermarket"}
},
```

**Comment by westnordost** [comment]
I created an issue at NSI to include this information also in the `nsi-id-presets.json` distribution, arguing that this is necessary information to correctly make use of the NSI. https://github.com/osmlab/name-suggestion-index/issues/10083

If this is included, then we can do the following:
0. add parsing support for this tag in the osmfeatures library
1. if for the selected preset the `"preserveTags"` is `"^name"`, then display the text field for the input of a name (pre-filled with the preset's name, but the user can change it).

**Comment by tiptoptom** [comment]
Sounds good. I would also display the text field if the name is not part of the NSI preset.
Here are some examples:
https://nsi.guide/index.html?t=brands&k=office&v=political_party

**Comment by westnordost** [comment]
[According to a short research](https://github.com/osmlab/name-suggestion-index/issues/10083#issuecomment-2484120690), StreetComplete (osmfeatures) and Go Map!! are the only data consumers which use the nsi-id-presets.json. I chatted with the author of Go Map!!, he agreed that maybe we should both migrate to nsi.json and get rid of the former.

**Comment by westnordost** [comment]
But nsi.json is not complete for what I need it. I instead implemented the linked issue
